### PR TITLE
6.2.lockupdate

### PR DESCRIPTION
**Pull request by jakelandis**
Introduce the Gemfile lock on the (new) 6.2 branch.  Diff the to commits in this PR to see the difference since 6.1 : https://github.com/elastic/logstash/pull/8972/commits/5f8fb6b004bd639856e5f1a0bf19aee03365b170

### CODE REVIEW

File: Gemfile.jruby-2.3.lock.release [hunk 1]
```diff
@@ -0,0 +1,774 @@
...
+      sequel
+    logstash-filter-json (3.0.5)
+      logstash-core-plugin-api (>= 1.60, <= 2.99)
+    logstash-filter-kv (4.0.3)
+      logstash-core-plugin-api (>= 1.60, <= 2.99)
+    logstash-filter-metrics (4.0.5)
+      logstash-core-plugin-api (>= 1.60, <= 2.99)
+      metriks
+      thread_safe
+    logstash-filter-multiline (3.0.4)
```
Comment (jsvd) [inline Gemfile.jruby-2.3.lock.release]: I think the lock file was created after running `rake test:install-core` before locking the dependencies. This installed a few gems used during testing which shouldn't be in the final lock file.
Whatever workflow we define as the one to lock the gems should depend on a `./gradlew clean` first.

Comment (jakelandis) [reply Gemfile.jruby-2.3.lock.release]: ah thanks, was wondering why there was such a large change.  rebuilding now.

**Comment by jakelandis** [comment]
@jsvd @guyboertje - lockfile updated to no longer include test gems.

**Comment by jsvd** [comment]
It's really interesting that we no longer list jar-dependencies and ruby-maven. It seems we use the same jruby, what changed?

**Comment by jakelandis** [comment]
> it's really interesting that we no longer list jar-dependencies and ruby-maven. It seems we use the same jruby, what changed?

@jsvd  - looks like the Java entry point removed them from logstash-core: https://github.com/elastic/logstash/commit/543b7221bd47afbde1f3c71059ff4be344990e80#diff-a70557553ce36f52ecddde61c83cb0f4 Note- they are still locked in the file, just not a dependency to logstash-core.

**Pull request closed without merging**